Emit emitter/buffers/allocated/delta and emitter/buffers/failed/delta

leventov opened the pull request:
@gianm gauge `emitter/buffers/failed` and `emitter/buffers/totalAllocated` metrics proved to be very inconvenient in practice: they are not additive across a fleet of nodes, are randomly reset when a JVM restarts (that may be completely independent from metrics emitting issues). Regarding [problem identification](https://github.com/apache/incubator-druid/pull/4973#issuecomment-338119125), `emitter/buffers/dropped` serves this well.

Currently in this PR I added `/delta` vs. `/total` metrics, but if there are no objections I would remove `/total` and so align `emitter/buffers/allocated` and `/failed` with `/dropped` and `/events/emitted` which are already delta-only.

gianm commented on java-util/src/main/java/org/apache/druid/java/util/metrics/HttpPostEmitterMonitor.java:
```diff
@@ -81,6 +96,24 @@ private void emitTimeCounterMetrics(ServiceEmitter emitter, ConcurrentTimeCounte
+  {
+    int newTotalAllocatedBuffers = httpPostEmitter.getTotalAllocatedBuffers();
+    int allocatedBuffersDelta = newTotalAllocatedBuffers - lastTotalAllocatedBuffers;
+    emitter.emit(builder.build("emitter/buffers/allocated/total", newTotalAllocatedBuffers));
```
This isn't compatible with the old name `emitter/buffers/totalAllocated`, but I think it's ok because we never documented the old names anyway.

QiuMM commented:
Got that, then it looks good to me.

leventov commented:
I removed `/total` and renamed `emitter/buffers/dropped` to `emitter/buffers/dropped/delta` and `emitter/events/emitted` to `emitter/events/emitted/delta` too.
